tests/unit/worker/test_submission_worker.py

```diff
@@ -79,8 +139,84 @@ def test_create_dir_as_python_package(self):
 
     def test_return_file_url_per_environment(self):
         returned_url = return_file_url_per_environment(self.url)
-        self.assertEqual(returned_url, "http://testserver/test/url")
+        self.assertEqual(returned_url, self.get_url_for_test_environment(self.url))
+
 
+@mock.patch("scripts.workers.submission_worker.importlib.import_module")
+class ExtractChallengeDataTest(BaseAPITestClass):
+    def setUp(self):
+        super(ExtractChallengeDataTest, self).setUp()
+
+        self.directory_path_patcher = mock.patch.multiple(
+            "scripts.workers.submission_worker",
+            CHALLENGE_DATA_DIR=self.CHALLENGE_DATA_DIR,
+            PHASE_DATA_BASE_DIR=self.PHASE_DATA_BASE_DIR,
+            PHASE_DATA_DIR=self.PHASE_DATA_DIR,
+            PHASE_ANNOTATION_FILE_PATH=join(self.PHASE_DATA_DIR, "{annotation_file}")
+        )
+        self.patcher_map = mock.patch(
+            "scripts.workers.submission_worker.PHASE_ANNOTATION_FILE_NAME_MAP",
+            {},
+        )
+        self.patcher_script_dict = mock.patch(
+            "scripts.workers.submission_worker.EVALUATION_SCRIPTS",
+            {},
+        )
+
+        self.phases = [self.challenge_phase]
+        self.annotation_file_name = os.path.basename(self.challenge_phase.test_annotation.name)
+        self.annotation_file_path = self.get_phase_annotation_file_path(
+            challenge_id=self.challenge.id,
+            phase_id=self.challenge_phase.id,
+            annotation_file=self.challenge_phase.test_annotation,
+        )
+        self.annotation_file_url = self.get_url_for_test_environment(
+            self.challenge_phase.test_annotation.url
+        )
+        self.challenge_data_directory = self.CHALLENGE_DATA_DIR.format(challenge_id=self.challenge.id)
+        self.challenge_import_string = self.CHALLENGE_IMPORT_STRING.format(challenge_id=self.challenge.id)
+        self.challenge_zip_file = join(self.challenge_data_directory, "challenge_{}.zip".format(self.challenge.id))
+        self.evaluation_script_url = self.get_url_for_test_environment(self.challenge.evaluation_script.url)
+
+    @mock.patch("scripts.workers.submission_worker.download_and_extract_file")
+    @mock.patch("scripts.workers.submission_worker.download_and_extract_zip_file")
+    def test_extract_challenge_data(self,
```

Comment:
@nikochiko can you also add tests for cases when exception is thrown

Reply:
@Ram81 That test case is already covered in the integration tests: https://github.com/Cloud-CV/EvalAI/blob/master/tests/integration/worker/test_submission_worker.py#L239
Do we need it here as well?

Reply:
Anyways, having all tests in one place makes more sense. Also coveralls says this is not covered: https://coveralls.io/builds/27483663/source?filename=scripts/workers/submission_worker.py#L257